deps/rabbit/src/rabbit.erl

```diff
@@ -921,6 +921,26 @@ do_run_postlaunch_phase(Plugins) ->
     rabbit_log_prelaunch:debug("== Postlaunch phase =="),
 
     try
+        %% Successful boot resets node maintenance state.
+        rabbit_log_prelaunch:debug(""),
+        rabbit_log_prelaunch:info("Resetting node maintenance status"),
+        _ = rabbit_maintenance:unmark_as_being_drained(),
+
+        %% Export definitions after all plugins have been enabled,
+        %% see rabbitmq/rabbitmq-server#2384
+        case rabbit_definitions:maybe_load_definitions() of
+            ok           -> ok;
+            DefLoadError -> throw(DefLoadError)
+        end,
+
+        %% The node is ready: mark it as such and log it.
+        %% NOTE: PLEASE DO NOT ADD CRITICAL NODE STARTUP CODE AFTER THIS.
+        %% TODO @dumbbell
+        %% rabbitmq/rabbitmq-server#2656
+        %% gotthardp/rabbitmq-email#42
+        rabbit_log_prelaunch:debug("Marking ~s as running", [product_name()]),
+        rabbit_boot_state:set(ready),
```

Comment:
We can't mark the broker as ready before plugins are started, that was one of the main issues the startup rewrite targetted. Otherwise, `rabbitmqctl wait` returns too early.

Perhaps that's `rabbit_direct:connect()` which must be improved to not depend on `rabbit:is_running()`.